Haptics/explainer.md

```diff
@@ -0,0 +1,205 @@
+               dRect.left - tRect.left,
+               dRect.top - tRect.top
+            );
+            const SNAP_DISTANCE = 40;
+            if (!snapped && distance < SNAP_DISTANCE) {
+               snapped = true;
+               // Snap the item visually
+               draggable.style.left = `${tRect.left}px`;
+               draggable.style.top = `${tRect.top}px`;
+               // Trigger the haptic feedback
+               if ("vibrate" in navigator) {
+                  navigator.vibrate(0, {waveform: 'align', intensity: 1});
+               }
+            } else if (snapped && distance >= SNAP_DISTANCE) {
+               snapped = false;
+            }
+         }
+      </script>
+   </body>
+</html>
+```
+
+## Alternatives considered
+
+- A similar declarative API without re-using `navigator.vibrate`
+    - Pros: the new API design is not limited by existing `navigator.vibrate` design. Developer will have better feature detection and the API can also return whether haptics is played successfully and can be async.
+    - Cons: New interface and added ecosystem complexity.
+
+- A pointer-event based API as previously defined in [explainer](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/HapticsDevice/explainer.md)
+    - Pros: Closely tie haptics to input events.
+    - Cons: New interface and added ecosystem complexity. Can be more cumbersome than a declarative method. 
+
+## Accessibility, Internationalization, Privacy, and Security Considerations
+
+### Privacy
+The current design does not expose means to query haptics-capable devices, available waveforms, or success in playing haptics to alleviate fingerprinting concerns.
+
+### Security
+The current design does not include permission gate for haptics but requires [sticky user activation](https://developer.mozilla.org/en-US/docs/Web/Security/User_activation). While there is potential concern for haptics spam, this can be mitigated by user agent stopping haptics when the user navigates away and there is no lasting user impact.
+
+## Reference for relevant haptics APIs
+
+This section provides reference to existing web and native haptics API to help inform the API design and platform supportability.
+
+Known platform-specific native haptics API: 
+
+- [Windows: InputHapticsManager](https://learn.microsoft.com/en-us/uwp/api/windows.devices.haptics?view=winrt-26100)
+- [macOS: NSHapticFeedbackManager](https://developer.apple.com/documentation/appkit/nshapticfeedbackmanager)
+- [iOS: Core Haptics](https://developer.apple.com/documentation/corehaptics)
+- [Android: VibrationEffect](https://developer.android.com/reference/android/os/VibrationEffect)
+
+Relevant web APIs:
+- [navigator.vibrate](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/vibrate)
+- [(In-progress) Gamepad Event-Driven Input API](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/6ea0b9190c7e8b0261323deaf060013d80a0b0ab/GamepadEventDrivenInputAPI/explainer.md)
```

Comment:
I'm not sure how the "Gamepad Event-Driven Input API" relates to haptics.  What did you have in mind?